cppcache/src/util/Log.hpp

```diff
@@ -182,315 +183,69 @@ class APACHE_GEODE_EXPORT Log {
+  ::apache::geode::client::Log::log(::apache::geode::client::LogLevel::Info, \
+                                    __VA_ARGS__)
 
-  static void fine(const std::string& message);
+#define LOGCONFIG(...)                                                         \
+  ::apache::geode::client::Log::log(::apache::geode::client::LogLevel::Config, \
+                                    __VA_ARGS__)
 
-  static void finer(const std::string& message);
+#define LOGFINE(...)                                                         \
+  ::apache::geode::client::Log::log(::apache::geode::client::LogLevel::Fine, \
+                                    __VA_ARGS__)
 
-  static void finest(const std::string& message);
-};
-}  // namespace client
-}  // namespace geode
-}  // namespace apache
+#define LOGFINER(...)                                                         \
+  ::apache::geode::client::Log::log(::apache::geode::client::LogLevel::Finer, \
+                                    __VA_ARGS__)
+
+#define LOGFINEST(...)                                                         \
+  ::apache::geode::client::Log::log(::apache::geode::client::LogLevel::Finest, \
+                                    __VA_ARGS__)
 
-#define LOGDEBUG                                  \
-  if (::apache::geode::client::LogLevel::Debug <= \
-      ::apache::geode::client::Log::logLevel())   \
-  ::apache::geode::client::LogVarargs::debug
-
-#define LOGERROR                                  \
-  if (::apache::geode::client::LogLevel::Error <= \
-      ::apache::geode::client::Log::logLevel())   \
-  ::apache::geode::client::LogVarargs::error
-
-#define LOGWARN                                     \
-  if (::apache::geode::client::LogLevel::Warning <= \
-      ::apache::geode::client::Log::logLevel())     \
-  ::apache::geode::client::LogVarargs::warn
-
-#define LOGINFO                                  \
-  if (::apache::geode::client::LogLevel::Info <= \
-      ::apache::geode::client::Log::logLevel())  \
-  ::apache::geode::client::LogVarargs::info
-
-#define LOGCONFIG                                  \
-  if (::apache::geode::client::LogLevel::Config <= \
-      ::apache::geode::client::Log::logLevel())    \
-  ::apache::geode::client::LogVarargs::config
-
-#define LOGFINE                                  \
-  if (::apache::geode::client::LogLevel::Fine <= \
-      ::apache::geode::client::Log::logLevel())  \
-  ::apache::geode::client::LogVarargs::fine
-
-#define LOGFINER                                  \
-  if (::apache::geode::client::LogLevel::Finer <= \
-      ::apache::geode::client::Log::logLevel())   \
-  ::apache::geode::client::LogVarargs::finer
-
-#define LOGFINEST                                  \
-  if (::apache::geode::client::LogLevel::Finest <= \
-      ::apache::geode::client::Log::logLevel())    \
-  ::apache::geode::client::LogVarargs::finest
+#define LOGDEBUG(...)                                                         \
```

Comment:
This change results in the loss of the LogLevel check to avoid the cost of executing functions that may make up the arguments to this log message. Now all arguments will be resolved and the log message executed regardless of the log message actually being loggable.

Reply:
Right.  Added the level check to the new versions of the macros, and removed it from the `Log::log` methods.